api/v1/helpers/verbs/doPatch.js

```diff
@@ -31,30 +33,38 @@ const httpStatus = require('../../constants').httpStatus;
 function doPatch(req, res, next, props) {
   const resultObj = { reqStartTime: new Date() };
   const requestBody = req.swagger.params.queryBody.value;
-  u.findByKey(props, req.swagger.params)
-  .then((o) => u.isWritable(req, o,
-    featureToggles.isFeatureEnabled('enforceWritePermission')))
-  .then((o) => {
-
-    // To avoid timeouts when patching samples; force the update, even if
-    // the value has not changed. Adding this to the "before update hook" does
-    // give the needed effect; so adding it here!!!.
-    if (props.modelName === 'Sample') {
-      o.changed('value', true);
+  let patchPromise;
+  if (featureToggles.isFeatureEnabled(constants.featureName) &&
+   props.modelName === 'Sample') {
+    const rLinks = requestBody.relatedLinks;
+    if (rLinks) {
+      u.checkDuplicateRLinks(rLinks);
```

Comment:
I thought we had to move validation only in case of redisStore and this is already handled for cases without redisStore. Atleast tests say that.